frontend/src/metabase/query_builder/components/view/ExecutionTime/ExecutionTime.tsx

```diff
@@ -5,11 +5,17 @@ import { Tooltip } from "metabase/ui";
 import { formatDuration } from "./utils";
 
 interface Props {
-  time: number;
+  time?: number;
 }
 
-export const ExecutionTime = ({ time }: Props) => (
-  <Tooltip label={t`Query execution time`}>
-    <span data-testid="execution-time">{formatDuration(time)}</span>
-  </Tooltip>
-);
+export const ExecutionTime = ({ time }: Props) => {
+  if (!time) {
```

Comment:
Maybe check for `if (time == null)`? So we could render `0`

Reply:
The `time` is explicitly `undefined` in case of an error. The response has the following shape:
```
{
  duration: number,
  error: string // or maybe object - I'd have to double check
}
```` 

In all other scenarios, the response seems to have an explicit `running_time`. Not sure if it's possible that it's ever `null`, since we don't have the type for it.

So I can add the check for null, just in case but we definitely need to leave the undefined check, IMO.
It makes sense not to display anything. We use the same approach for "Showing ___ rows", after all.

Reply:
@qnkhuat helped me check the type for running_time and it's set to not null.
![image](https://github.com/user-attachments/assets/c6bf15b8-a9c0-403f-a08c-1880f69f9e8f)

Reply:
I'm pretty sure @ranquild meant the `=== undefined` case but `==` made it ambiguous.

https://www.notion.so/metabase/56f6f59da3b647de89acb28ea9ebc424?v=cca33933b6244a7a94f256d1c110ce09&p=5e63b631c8c545a9bbd272e54deeeebe&pm=s ;)

Reply:
In that case, here's an explicit check for `undefined`.
This also simplifies the logic as I don't have to guard against 0.

Reply:
I meant that we would ignore `0` this way